packages/amplify-category-geo/src/service-walkthroughs/resourceWalkthrough.ts

```diff
@@ -1,41 +1,106 @@
 import { AccessType, DataProvider, ResourceParameters } from "../service-utils/resourceParams";
 import { apiDocs, ServiceName } from "../service-utils/constants";
-import { $TSContext, open } from "amplify-cli-core";
-import { printer, prompter } from 'amplify-prompts';
+import { prompter, printer, byValue, byValues } from 'amplify-prompts';
+import { $TSContext } from "amplify-cli-core";
 
-export async function resourceAccessWalkthrough<T extends ResourceParameters>(
+export async function resourceAccessWalkthrough<T extends ResourceParameters & { groupPermissions: string[] }>(
+    context: $TSContext,
     parameters: Partial<T>,
     service: ServiceName
 ): Promise<Partial<T>> {
-    const accessChoices = [
-        { name: 'Authorized users only', value: AccessType.AuthorizedUsers },
-        { name: 'Authorized and Guest users', value: AccessType.AuthorizedAndGuestUsers }
-    ];
-    let accessTypeDefaultIndex = 0;
-    if (parameters.accessType === AccessType.AuthorizedAndGuestUsers) {
-        accessTypeDefaultIndex = 1;
+    let permissionSelected = 'Auth/Guest Users';
+    const LearnMore = 'Learn more';
+    const userPoolGroupList = context.amplify.getUserPoolGroupList();
+
+    if (userPoolGroupList.length > 0) {
+        do {
+            if (permissionSelected === LearnMore) {
+                printer.info('');
+                printer.info(
+                    'You can restrict access using CRUD policies for Authenticated Users, Guest Users, or on individual Group that users belong to in a User Pool. If a user logs into your application and is not a member of any group they will use policy set for “Authenticated Users”, however if they belong to a group they will only get the policy associated with that specific group.',
+                );
+                printer.info('');
+            }
+            let defaultPermission = 'Auth/Guest Users';
+            if (parameters.accessType === AccessType.CognitoGroups) {
+                defaultPermission = 'Individual Groups'
+            }
+            else if (parameters.groupPermissions &&
+                parameters.groupPermissions.length > 0 &&
+                parameters.accessType !== AccessType.CognitoGroups) {
+                defaultPermission = 'Both'
+            }
+            permissionSelected = await prompter.pick<'one', string>(
+                `Restrict access by?`,
+                ['Auth/Guest Users', 'Individual Groups', 'Both', LearnMore],
```

Comment:
Why not make this a multi-select prompt rather than have a "both" option?

Reply:
The behavior here aligns with other categories and so something the Amplify users must be already familiar with.